addons/pos_viva_wallet/static/src/app/payment_viva_wallet.js

```diff
@@ -0,0 +1,199 @@
+        const line = this.pos.get_order().selected_paymentline;
+
+        var data = {
+            "sessionId": line.sessionId,
+            "cashRegisterId": this.pos.get_cashier().name
+        };
+        try {
+            const resp = this.pos.env.services.orm.silent.call(
+                "pos.payment.method",
+                "viva_wallet_send_payment_cancel",
+                [[this.payment_method.id], data]
+            );
+            if (resp.error) {
+                throw resp.error;
+            }
+        } catch (error) {
+            this._showError(error, 'Unable to cancel payment');
+            return false;
+        }
+        line.set_payment_status("retry");
+        return true;
+    }
+
+    /**
+     * This method is called from pos_bus when the payment
+     * confirmation from Viva Wallet is received via the webhook and confirmed in the retrieve_session_id.
+     */
+    async handleVivaWalletStatusResponse() {
+        var line = this.pending_viva_wallet_line();
+        const notification = await this.env.services.orm.silent.call(
+            "pos.payment.method",
+            "get_latest_viva_wallet_status",
+            [[this.payment_method.id]]
+        );
+
+        if (!notification) {
+            this._handle_odoo_connection_failure();
+            return;
+        }
+
+        const isPaymentSuccessful = this.isPaymentSuccessful(notification);
+        if (isPaymentSuccessful) {
+            this.handleSuccessResponse(line, notification);
+        } else {
+            this._show_error(
+                sprintf(_t("Message from Viva Wallet: %s"), notification.get("message"))
+            );
+        }
+
+        // when starting to wait for the payment response we create a promise
+        // that will be resolved when the payment response is received.
+        // In case this resolver is lost ( for example on a refresh ) we
+        // we use the handle_payment_response method on the payment line
+        const resolver = this.paymentLineResolvers?.[line.cid];
+        if (resolver) {
+            resolver(isPaymentSuccessful);
+        } else {
+            line.handle_payment_response(isPaymentSuccessful);
+        }
+    }
+
+    isPaymentSuccessful(notification) {
+        return (
+            notification &&
+            notification.sessionId ==
+                this.pending_viva_wallet_line().sessionId &&
+            notification.success
+        );
+    }
+
+    waitForPaymentConfirmation() {
+        return new Promise((resolve) => {
+            this.paymentLineResolvers[this.pending_viva_wallet_line().cid] = resolve;
+        });
+    }
+
+    handleSuccessResponse(line, notification) {
+        line.transaction_id = notification.transactionId;
+        line.card_type = notification.applicationLabel;
+        line.cardholder_name = notification.FullName || "";
+    }
+
+    _show_error (msg, title) {
+        if (!title) {
+            title = "Viva Wallet Error";
```

Comment:
```suggestion
            title = _t("Viva Wallet Error");
```

Translate. Don't forget to import `_t`.